docs/source/getting-started/what-is-jupyterhub.md

```diff
@@ -0,0 +1,465 @@
+# What is Jupyter and JupyterHub?
+
+JupyterHub is not what you think it is.  Most things you think are
+part of JupyterHub are actually handled by some other component, for
+example the spawner or notebook server itself, and it's not always
+obvious how the parts relate.  The knowledge contained here hasn't
+been assembled in one place before, and is essential to understand
+when setting up a sufficiently complex Jupyter(Hub) setup.
+
+This document was originally written to assist in debugging: very
+often, the actual problem is not where one thinks it is and thus
+people can't easily debug.  In order to tell this story, we start at
+JupyterHub and go all the way down to the fundamental components of
+Jupyter.
+
+In this document, we occasionally leave things out or bend the truth
+where it helps in explanation, and give our explanations in terms of
+Python even though Jupyter itself is language-neutral.  The "(&)"
+symbol highlights important points where there is more.
+
+This guide is long, but after reading it you will be know of all major
+components in the Jupyter ecosystem and everything else you read
+should make sense.
+
+
+
+## Just what is Jupyter?
+
+Before we get too far, let's remember what our end goal is.  A
+**Jupyter Notebook** is really nothing more than a Python(&) process
+which is getting commands from a web browser and displaying the output
+via that browser.  What the process actually sees can roughly like
+getting commands on standard input(&) and writing to standard
+output(&).  There is nothing intrinsically special about this process
+- it can do anything a normal Python process can do, and nothing more.
+The **Jupyter kernel** handles capturing output and converting things
+such as graphics to a form usable by the browser.
+
+Everything we explain below is building up to this, going through many
+different layers which give you many ways of customizing how this
+process runs.  But this process is not *too* special.
+
+
+
+## JupyterHub
+
+**JupyterHub** is the central piece that provides multi-user
+login. Despite this, the end user only briefly interacts with
+JupyterHub and most of the actual Jupyter session does not relate to
+the hub at all: the hub mainly handles authentication and spawning the
```

Comment:
```suggestion
the hub at all: the hub mainly handles authentication and creating (jargon: spawning) the
```
What do you think of something like this to "define" the jargon used in JupyterHub land? Trying to model it on how one should define three letter acronyms (TLA) before using them later. In the next sentence we can now talk about why TLAs appear everywhere.

Reply:
Very good idea!  I especially like this syntax of defining the jargon.